corehq/apps/hqwebapp/static/hqwebapp/js/base_ace.js

```diff
@@ -1,15 +1,13 @@
 hqDefine('hqwebapp/js/base_ace', [
     'jquery',
     'ace-builds/src-min-noconflict/ace',
+    'ace-builds/src-min-noconflict/mode-json',
 ], function (
     $,
-    ace
+    ace,
+    jsonMode
 ) {
 
-    if (!ace.config.get('basePath')) {
```

Comment:
hey, Did you try loading this on the browser and not just in the `build_requirejs` script. I have added this piece of code here because the ace was not able to find its dependencies from the CDN As its base path was coming unset when it is loaded from the browser. I remember it worked the local machine but was failing on the staging because of what I briefed above. It will be best to also test this change running on staging.

Reply:
Yes, I debugged this using the browser.

I also found that it worked locally when static media was served by Django, but I was able to reproduce the bug by using `build_requirejs` [as described here](https://github.com/dimagi/commcare-hq/blob/master/docs/js-guide/migrating.md#troubleshooting-the-requirejs-build-process).

I will test this on staging, thanks.

Reply:
@rohit-dimagi the problem is that `requirejs.s.contexts._.config.paths` is not a public API as far as I can tell. It would be ideal to find a solution that does not need to reach into requirejs internals.

@kaapstorm if this solution does not work you might try [this](http://www.howtobuildsoftware.com/index.php/how-do/bSWF/%E2%80%9C). I'm not convinced it will work, but worth a try?